index.rst

```diff
@@ -48,10 +48,109 @@
+Informed by the data management git repositories, I suggest the following:
+
+- ``notebooks/wip`` -- A directory that holds notebooks that are works in progress.
+  These need not adhere to any specific quality specifications, but should have enough documentation to be readable by someone else.
+- ``notebooks/prod`` -- Notebooks meant to adhere to specific quality guidelines.
+  Notebooks cannot be added to ``prod`` from ``WIP`` without a review.
+- ``bin`` -- A place to put command line tools.
+    Command line tools are constructed from python modules as much as possible to improve test coverage.
+    I.e. any code logic should be factored into functions/methods in modules in the ``python`` directory.
+- ``python`` -- Python library code goes here.
+- ``tests`` -- A place for unit tests of the modular python code.
+  
+These repositories will be ``eups`` packages so that we can use all the existing infrastructure.
+
+Notes on notebook review
+------------------------
+
+Notebooks should be reviewed by someone other than the original author.
+The review should look for several things:
+
+- The notebook should adhere to a standard templated format (not defined here).
+  At the very least, the notebook should state the end product from the notebook, the responsible author, and the version of the stack it is compatible with (perhaps computed by the CI system).
+- When possible, the notebook should run sequentially from top to bottom.
+- The reviewer should also look at the markdown cells for accuracy.
+- Any duplication of code between this and another notebook should be identified and factored out into a module.
+- Reviews should attempt to promote accepted idioms for common tasks.
+
+General suggestions
+-------------------
+
+Notebook repositories should be testable.
+We do not have a notebook continuous integration system yet, but when it exists, these repositories should regularly have both unit tests and notebook tests run.
+This only applies to notebooks in ``notebooks/prod``.
+
+Be aggressive about moving common code into python modules rather than letting multiple copies bit rot in separate notebooks.
+
+Have a concrete plan for what to do about notebooks that do not pass testing.
+At some point they should be moved back to ``notebooks/WIP``, but whether that is immediate or there is a grace period for the author to fix the notebook should be up to the repository owner.
+
+Whether notebooks are cleared before committing, will depend on the intentions of the team using the notebook repository.
+
+- Clearing outputs will reduce the overall size, make the diff cleaner, and makes the notebook clean for the next person to execute
+- Leaving outputs in the committed notebook will save time since the notebook does not need to be executed to see plots etc.
+  Leaving outputs also allows for robust testing since ``nbeval`` can compare outputs from the tested notebook match those of the input notebook.
+
```

Comment:
For DM testing, see DMTR-201 and  DMTR-161, a convention that has been adopted is to have a directory called 'notebooks' and a directory called 'executions'. The latter is used to store notebooks with outputs, if desired.